### PR TITLE
Restructure and Add Unit tests

### PR DESCRIPTION
**Pull request by fahad-israr**
Related Issue: https://github.com/starfixdev/starfix/issues/78

@maxandersen I've tried her to restructure some portions so that I can add unit Tests specifically considered the corner cases we have resolved before so that we may not break them in future.

### CODE REVIEW

**Review by maxandersen — APPROVED** [review]

**Review by maxandersen — CHANGES_REQUESTED** [review]
except the picocli annotation processor changes this looks ok. 

please remove that change as its discussed in #85

**Comment by fahad-israr** [comment]
> except the picocli annotation processor changes this looks ok.
> 
> please remove that change as its discussed in #85

Sure 👍

File: cli/pom.xml [hunk 1]
```diff
@@ -61,6 +61,24 @@
...
     <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <!-- annotationProcessorPaths requires maven-compiler-plugin version 3.5 or higher -->
+        <version>${maven-compiler-plugin-version}</version>
+        <configuration>
+            <annotationProcessorPaths>
+              <path>
+                  <groupId>info.picocli</groupId>
```
[the remark this answers is not in the document]
Comment (fahad-israr) [reply cli/pom.xml]: Removed .

**Comment by fahad-israr** [comment]
> except the picocli annotation processor changes this looks ok.
> 
> please remove that change as its discussed in #85

done !
